R/feature-availability-profiler.R

```diff
@@ -0,0 +1,279 @@
+#' by hour, after a particular event (like patient admit)
+#' 
+#' @param df A dataframe
+#' @param startDateColumn Optional string of the column name, representing the 
+#' date of the starting event of interest (e.g., patient admit)
+#' @param lastLoadDateColumn Optional string of the column name, representing
+#' the date the row was loaded into the final dataset (i.e., via daily ETL)
+#' @param plotProfiler Default is TRUE. Whether to plot profiler results 
+#' @return a list, that has as many vectors as columns in the original 
+#' dataframe, with each vector holding the percentage full for each hour
+#'
+#' @export
+#' @references \url{http://healthcare.ai}
+#' @seealso \code{\link{healthcareai}}
+#' @examples 
+#' df <- data.frame(a = c(2,1,3,5,4,NA,7,NA),
+#'                  b = c(0.7,-2,NA,-4,-5,-6,NA,NA),
+#'                  c = c(100,300,200,NA,NA,NA,NA,500),
+#'                  d = c(407,500,506,504,NA,NA,NA,405),
+#'                  admit = c('2012-01-01 00:00:00','2012-01-01 00:00:00',
+#'                            '2012-01-01 12:00:00','2012-01-01 12:00:00',
+#'                            '2012-01-02 00:00:00','2012-01-02 00:00:00',
+#'                            '2012-01-02 12:00:00','2012-01-02 12:00:00'),
+#'                  loaded = c('2012-01-03 00:00:00','2012-01-03 00:00:00',
+#'                             '2012-01-03 00:00:00','2012-01-03 00:00:00',
+#'                             '2012-01-03 00:00:00','2012-01-03 00:00:00',
+#'                             '2012-01-03 00:00:00','2012-01-03 00:00:00'))
+#' str(df)
+#' head(df)
+#' 
+#' d <- featureAvailabilityProfiler(df = df,
+#'                                  startDateColumn = 'admit',
+#'                                  lastLoadDateColumn = 'loaded')
+#' d # Look at the data in the console                                
+
+featureAvailabilityProfiler = function(
+  df,
+  startDateColumn='AdmitDTS',
+  lastLoadDateColumn='LastLoadDTS',
+  plotProfiler=TRUE) {
+  
+  # Error handling
+  # Make sure that it's a dataframe
+  if (missing(df) || class(df) != 'data.frame' ) {
+    stop('Please specify a dataframe')
+  }
+  
+  # With at least 3 columns
+  if (dim(df)[2] < 3) {
+    stop('Dataframe must be at least 3 columns')
+  }
+  
+  # Check that date columns are in dataframe
+  if (!(startDateColumn %in% colnames(df))) {
+    stop(paste0(startDateColumn,' is not in your dataframe.',
+         ' Please carefully specify the startDateColumn'))
+  }
+  
+  if (!(lastLoadDateColumn %in% colnames(df))) {
+    stop(paste0(lastLoadDateColumn,' is not in your dataframe.',
+         ' Please carefully specify the lastLoadDateColumn'))
+  }
+  
+  # Convert specified date columns to dates, if necessary
+  if (class(df[[startDateColumn]])[1] != 'POSIXct') {
+    tryCatch(
+      df[[startDateColumn]] <- as.POSIXct(df[[startDateColumn]]),
+      error = function(e) {
+      e$message <- paste0(startDateColumn, " may not be a date column. \n", e)
```

Comment:
Could this error message give more guidance as to what date type should be? YYYY-MM-DD?